blogArticle.md

```diff
@@ -0,0 +1,590 @@
+
+- the methods `userSignup` and `userLogin` that will be used further on.
+At the moment, there is no form validation of any kind.
+The only thing they do is call the `Action` method from `react-native-router-flux` and make a scene to transition to the Homepage scene
+
+- a render method, which will display two text inputs (whose values are already bounded to our Component's state) and two buttons, each one bound to the `userSignup` and `userLogin` methods.
+
+Moving forward and defining the `Homepage` class:
+
+```javascript
+// app/routes/Homepage.js
+
+import React, {Component} from 'react';
+import {Alert, Image, Text, TouchableOpacity, View} from 'react-native';
+import {Actions} from 'react-native-router-flux';
+import styles from './styles';
+
+class HomePage extends Component {
+
+  getProtectedQuote() {
+    Alert.alert("We will print a Chuck Norris quote")
+  }
+
+  userLogout() {
+    Actions.Authentication();
+  }
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <Image
+          source={require('../images/chuck_norris.png')}
+          style={styles.image}
+        />
+
+        <TouchableOpacity
+          style={styles.buttonWrapper}
+          onPress={this.getProtectedQuote}
+        >
+          <Text style={styles.buttonText}>
+            Get Chuck Norris quote!
+          </Text>
+        </TouchableOpacity>
+
+        <TouchableOpacity
+          style={styles.buttonWrapper}
+          onPress={this.userLogout}
+        >
+          <Text style={styles.buttonText} >
+            Log out
+          </Text>
+        </TouchableOpacity>
+      </View>
+    );
+  }
+}
+
+export default HomePage;
+```
+
+Again, let's go through the details of what we just wrote.
+This time, we have:
+
+- a `HomePage` class with no constructor defined, because our component is stateless
+
+- a `getProtectedQuote` method, that will be responsible for communicating with an API's protected route to recover a funny Chuck Norris quote.
+At the moment it just shows an alert popup with a title.
+
+- an `userLogout` method, that redirects the user to the Authentication scene for now.
+
+- a render method, which will display an image and two buttons, each one bound to the `getProtectedQuote` and `userLogout` methods
+
+Both our scenes import basic style properties from an exterior file, which can be seen on our [this project's repository](https://github.com/febeck/ReactNativeAuth/blob/master/app/routes/styles.js).
+
+#### Authenticating the user
+
+The first step is to create a method that will save the received id token from the API in the `AsyncStorage`, the equivalent of the the browser's `LocalStorage`.
+
+The reason why the token will be stored is to be able to recover this data every time we have to do an API request for a protected route and further on to create the persistent user session.
```

Comment:
Je formulerais plutôt:

The reason the token needs to be stored is to be able to recover it every time we have to call a protected API route and later on to create the persistent user session.